### ISSUE TITLE
Big regression for card icons! (for some text, several caracters are displayed)

### ISSUE THREAD

**Issue by Altonss** [CatimaLoyalty/Android#839]
Now emojis are now supported, which is super cool!
But I noticed a severe regression for some card where the name is "cov" or "Atelier".

**Comment by TheLastProject** [comment]
Seems fine here: 
![image](https://user-images.githubusercontent.com/1885159/158063385-6479ce8a-fb99-4192-b19e-ba9d487d3042.png)

I'm guessing this is Android 5 specific again. It only happens for some words like 'cov" or "Atelier" but not all words?

**Comment by Altonss** [comment]
Yes exactly!

**Comment by Altonss** [comment]
Could some kind of debug log help?

**Comment by TheLastProject** [comment]
I can confirm this in an emulator and can confirm in logging it goes wrong here:

https://github.com/CatimaLoyalty/Android/commit/d9f0ea8346b24739c51eda8554403b10f747b6c2#diff-20d79f47432dd2df76984aa6ad72fb214d02bb1f61d7e6d202fd81852ec0b26fR73-R84

```
D/LetterBitmap: using sequence A to render first char which has length 1
D/LetterBitmap: using sequence At to render first char which has length 2
D/LetterBitmap: using sequence Ate to render first char which has length 3
```

Why Android is telling us that values like `cov`, `At` or `Ate` are a single-character sequence (like emojis are)... I have no clue. But given that [according to a Google bugracker](https://issuetracker.google.com/issues/170587912?pli=1) this code uses different functions on API 23 and up (Android 6.0) and I don't see it on my Android 11 phone, I assume this is an Android 5 specific bug.

I am not sure what the correct way to solve this is. A hacky method would be to brute-force characters and see which combinations fail and explicitly return the right value for those on Android 5. Given there are only 53 Android 5 users using Catima (according to Google Play Console), which is about 0.6% of the almost 9000 users, I feel like doing something hacky like this might be most sane.

**Comment by Altonss** [comment]
I understand your point of view!

**Comment by TheLastProject** [comment]
After some more emulator testing I can confirm only Android 5 is affected, Android 6 is already working fine

**Comment by TheLastProject** [comment]
Okay, so turns it this is not 100% Android 5. Apparently "ffi" is also detected and rendered in a special way even up to Android 11.

This probably needs a bit of extra logic to only check for glyphs if the characters fall in the emoji spectrum

**Comment by Altonss** [comment]
Thanks a lot for the extra testing! I'm relieved to see that it's not just a problem with my old android 5, but I don't really understand the origin of the problem ^^

**Comment by TheLastProject** [comment]
As I understand it the issue is that the "emoji detection" code doesn't only detect emojis, it also detects other unicode character sequences that we don't want to treat as special (I don't think any user cares about ffi being rendered into a single "character", way too technical)

**Comment by Altonss** [comment]
Isn't there a way to detect emojis or special unicode caracters vs. "classic" caracters (letters, numbers...)?

**Comment by TheLastProject** [comment]
Figuring out how to detect if the sequence is an emoji or not seems pretty much like the fix to this issue, yeah. There probably is some way to do that.

**Comment by TheLastProject** [comment]
I just realized a much easier solution: just check if all characters fall between a-Z0-9 and don't symbolize them if they do. 5 minute fix :D

**Comment by TheLastProject** [comment]
Should be fixed in 34639f2a2e77ee9d863f50e555070782ef6b4866.
